compose/service.py

```diff
@@ -451,16 +452,29 @@ def start_container(self, container):
     def connect_container_to_networks(self, container):
         connected_networks = container.get('NetworkSettings.Networks')
 
-        for network, aliases in self.networks.items():
+        for network, netdefs in self.networks.items():
+            aliases = netdefs.get('aliases', [])
+            ipv4_address = netdefs.get('ipv4_address', None)
+            ipv6_address = netdefs.get('ipv6_address', None)
             if network in connected_networks:
                 self.client.disconnect_container_from_network(
                     container.id, network)
 
-            self.client.connect_container_to_network(
-                container.id, network,
-                aliases=list(self._get_aliases(container).union(aliases)),
-                links=self._get_links(False),
-            )
+            if docker.version_info >= (1, 8, 0):
```

Comment:
We're going to set docker-py to a minimum version of 1.8.0, so we don't need this if/else branch.

For now you can set it to a git sha that contains the necessary commit in `requirements.txt`.